lib/byebug/breakpoint.rb

```diff
@@ -31,12 +31,13 @@ def self.add(file, line, expr = nil)
     end
 
     #
-    # Removes a breakpoint
+    # Removes a breakpoint and returns the breakpoint information
     #
     # @param id [integer] breakpoint number
     #
     def self.remove(id)
-      Byebug.breakpoints.reject! { |b| b.id == id }
+      bi = Byebug.breakpoints.find_index { |b| b.id == id }
+      bi ? Byebug.breakpoints.delete_at(bi) : nil
```

Comment:
@deivid-rodriguez Line 34 suggests what I was thinking of, that change returns the breakpoint object and this has other information about the breakpoint such as whether it had been enabled or disabled,  if there was a condition attached and so on. 

However it looks like the only use of this, the delete command doesn't make use of that additional information. So sure, the only essential part  is in delete.rb and this doesn't need to be modified.